**Simple ANN(Artifical Neural Network)** means the neural network only have two layers
* input layer
* output layer

**Here is the following steps to create simple ANN**
1. Install tensorflow and necessaries libraries
2. Load Dataset
3. Data Preprocessing(handle missing and dupliate values, remove outliers , scaling and encoding etc)
4. Select Features columns and Target Columns(X & Y)
5. Spliting Dataset(Train_Test_Split)
6. Standerdize data if you did not standardize data in step 3.
7. Define Model Type (use sequential method which is stack the layers)
   * Add Layers(Input_Layer,Hidden_Layer,Output_Layer)
   * Add 1 hidden layers with 10 neurons and use 'ReLU' activation function for non linear processing.then add an output layer with 1 neuron using sigmoid activation function which is suitable for binary classification.
8. Compile Model(Loss_Function(binary_crossentropy),optimizer='adam',metrices[accuracy])
9. Training Model(Epochs and Batch Size)
  * Epochs mean number of iteration you want to perform on data
  * Batch Size means Numbers of Samples
10. Evaluate Model



# **Step 1: Import libraries**

In [ ]:
import pandas as pd
import seaborn as sns
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
import tensorflow as tf
import warnings
warnings.filterwarnings('ignore')

# **Step 2: Load Dataset**

In [ ]:
dataset=pd.read_csv('/content/Titanic-Dataset.csv')

# **Step 3: Data Preprocessing**

In [ ]:
dataset.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [ ]:
missing_values=dataset.isnull().sum()
missing_values

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

In [ ]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [ ]:
dataset['Embarked'].fillna(dataset['Embarked'].mode()[0],inplace=True)
dataset.drop(['Age','Cabin'],axis=1,inplace=True)

In [ ]:
dataset.isnull().sum()

PassengerId    0
Survived       0
Pclass         0
Name           0
Sex            0
SibSp          0
Parch          0
Ticket         0
Fare           0
Embarked       0
dtype: int64

In [ ]:
dataset.duplicated().sum()

np.int64(0)

In [ ]:
#Categorical Columns
categorical_column=dataset.select_dtypes(include='object').columns
categorical_column

Index(['Name', 'Sex', 'Ticket', 'Embarked'], dtype='object')

In [ ]:
#Encoding Categorical Columns
dataset=pd.get_dummies(dataset,columns=categorical_column,drop_first=True)

# **Step 4: Feature Selection**

In [ ]:
# x and y
X=dataset.drop(['PassengerId','Survived'],axis=1)
y=dataset['Survived']

# **Step 5: Train Test Split**

In [ ]:
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.2,random_state=42)

# **Step 6: Scaling Data**

In [ ]:
#scaling data
scaler=StandardScaler()
X_train=scaler.fit_transform(X_train)
X_test=scaler.transform(X_test)

# **Step 7 : Define Model Type & Add Layers**

In [ ]:
#Create Layers
input_layer=tf.keras.layers.Dense(10,activation='relu',input_shape=(X_train.shape[1],))
output_layer=tf.keras.layers.Dense(1,activation='sigmoid')
#add layers into model(Sequential)
model=tf.keras.Sequential([input_layer,output_layer])
model.summary()

Model: "sequential_3"

┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━┓
┃ Layer (type)                    ┃ Output Shape           ┃       Param # ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━┩
│ dense_6 (Dense)                 │ (None, 10)             │        15,780 │
├─────────────────────────────────┼────────────────────────┼───────────────┤
│ dense_7 (Dense)                 │ (None, 1)              │            11 │
└─────────────────────────────────┴────────────────────────┴───────────────┘

 Total params: 15,791 (61.68 KB)

 Trainable params: 15,791 (61.68 KB)

 Non-trainable params: 0 (0.00 B)

In [ ]:
X_train.shape

(712, 1577)

# **Step 8: Compile Model**

In [ ]:
model.compile(optimizer='adam',loss='binary_crossentropy',metrics=['accuracy'])

# **Step 9: Training Model**

In [ ]:
train=model.fit(X_train,y_train,epochs=10,batch_size=32,verbose=1)

Epoch 1/10
23/23 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.9960 - loss: 0.0183
Epoch 2/10
23/23 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.9927 - loss: 0.0205
Epoch 3/10
23/23 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.9926 - loss: 0.0227
Epoch 4/10
23/23 ━━━━━━━━━━━━━━━━━━━━ 0s 7ms/step - accuracy: 0.9947 - loss: 0.0210
Epoch 5/10
23/23 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.9955 - loss: 0.0200
Epoch 6/10
23/23 ━━━━━━━━━━━━━━━━━━━━ 0s 9ms/step - accuracy: 0.9921 - loss: 0.0229
Epoch 7/10
23/23 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.9915 - loss: 0.0177
Epoch 8/10
23/23 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.9954 - loss: 0.0188
Epoch 9/10
23/23 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.9931 - loss: 0.0229
Epoch 10/10
23/23 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.9947 - loss: 0.0148


accuracy: 0.9947 & loss: 0.0148 of training data


# **Step 10: Evaluate Model**

In [ ]:
#checking loss and accuracy of testing data
loss,Accuracy=model.evaluate(X_test,y_test)
print(f'Loss: {loss}')
print(f'Accuracy: {Accuracy}')

6/6 ━━━━━━━━━━━━━━━━━━━━ 0s 11ms/step - accuracy: 0.7049 - loss: 0.6282 
Loss: 0.5980105996131897
Accuracy: 0.7094972133636475
